packages/dds/tree/src/test/snapshots/summary/Compressed/v1/optional-field-scenarios-final.json

```diff
@@ -330,13 +330,13 @@
                                             "change": {
                                               "m": [
                                                 [
-                                                  1,
```

Comment:
@yann-achard-MS - do these snapshot changes make sense to you? From what I can tell, the IDs being generated for builds (I think) are different, though they are still consistent (e.g. all 0s became 1s and all 1s became 3s). So I guess I am burning some IDs now, and I'm not sure why, though it shouldn't be incorrect...

Reply:
@alex-pardes gave me some guidance that it probably has to do with resetting the change family's id allocator (when calling `editor.exitTransaction`. However, I still haven't figured out what is different about the sequencing before vs. after this PR.

Reply:
I've investigated and this should be fine. Here's what is happening.

Before this PR, transactions happen on the checkout's original branch. So, as edits occur, the ID counter in ModularChangeFamily increments with each new edit. However, anytime that a transaction happens - which is still happening on the original branch - it will reset the ID counter back down to 0 when the transaction closes.

After this PR, that reset doesn't happen anymore because the transaction always happens on its own branch and it does not affect the original branch. So, the original branch will have its IDs just continue to grow rather than being inadvertently reset. That's why we have some IDs that are a bit larger than they were before here.

Reply:
Thanks for figuring that out.